opencog/atoms/value/LinkValue.cc

```diff
@@ -40,6 +40,23 @@ bool LinkValue::operator==(const Value& other) const
 	return true;
 }
 
+bool LinkValue::operator<(const Value& other) const
+{
+	if (LINK_VALUE != other.get_type())
+		return LINK_VALUE < other.get_type();
+
+	const LinkValue* lov = (const LinkValue*) &other;
+
+	return _value < lov->_value;
```

Comment:
huh??? is there some existing `operator<(std::vector<Value>&)` ??? How can this possibly work?

Reply:
Well std::vector has templates for the operator< so as long as you can compare the elements you can compare vectors. https://en.cppreference.com/w/cpp/container/vector/operator_cmp

Reply:
Ah, well, now that you say this, I suppose that it's true; of course they'd provide something like this.

For readability, it would be nicer to write
```
    return _value.operator<(lov->_value);
```
because that naked less-than seems confusing, a least to me, as a veteran c++ programmer...

Reply:
I guess having worked so much with Haskell I am just more used to things like that. I find it annoying when I can't print a vector automatically when I know how to print an element.

I mean I can change that no problem just don't really see the difference between the 2 version other than yours having more characters.